module-apps/application-call/windows/CallWindow.hpp

```diff
@@ -74,10 +57,17 @@ namespace gui
         void destroyInterface() override;
         status_bar::Configuration configureStatusBar(status_bar::Configuration appConfiguration) override;
 
-        bool handleDigit(const uint32_t digit);
         void connectTimerOnExit();
-        void runCallTimer();
-        void stopCallTimer();
+        void refreshWindow() override;
+
+        void updateDuration(const UTF8 &text, bool isVisible = true) override;
+        void setNavBarForActiveCall() override;
+        void setNavBarForIncomingCall() override;
+        void clearNavBar() override;
+        void showIncomingCallLayout(bool isValidCallerId) override;
+        void showActiveCallLayout() override;
+        void showCallEndedLayout() override;
+        void updateNumber(const UTF8 &text) override;
```

Comment:
I personally prefer to have functions in the view named `showSomething` as `set` or `update` assumes that there was a previous state - and the view shouldn't mind state too much. 
Then we i.e. have flow like that:
`window.onInput() -> presenter.handleUIEnter() -> ui.showNumber()`
which might be a bit easier to read as we immediately know that UI only shows what the presenter handled before. ( I hope it was explanatory enough)